generators/generator-base.js

```diff
@@ -112,6 +112,14 @@ module.exports = class JHipsterBaseGenerator extends PrivateBase {
             this.config.set(this.options.localConfig);
         }
 
+        this.option('skip-generated-flag', {
+            desc: 'Skip adding a GeneratedByJhipster annotation to all generated java classes and interfaces',
+            type: Boolean,
+        });
```

Comment:
Move the option register to before `if (this.options.help)` otherwise it won't be shown at help and commander will not be aware of this option.